src/source_frames.rs

```diff
@@ -251,18 +266,47 @@ fn drop_frame_tc_adjustment(sections: TimecodeSections, rate: Framerate) -> Fram
     Ok(-adjustment)
 }
 
-fn parse_feet_and_frames_str(matched: regex::Captures) -> FramesSourceResult {
+fn parse_feet_and_frames_str(
+    matched: regex::Captures,
+    given_format: Option<FilmFormat>,
+) -> FramesSourceResult {
     // If we got a match, these groups had to be present, so we can unwrap them.
+
     let feet = timecode_parse::convert_tc_int(matched.name("feet").unwrap().as_str(), "feet")?;
-    let mut frames =
+    let frames =
         timecode_parse::convert_tc_int(matched.name("frames").unwrap().as_str(), "frames")?;
 
+    let perfs_str = matched.name("perf");
+
     // Get whether this value was a negative timecode value.
     let is_negative = matched.name("negative").is_some();
-
-    frames += feet * FRAMES_PER_FOOT;
-    if is_negative {
-        frames = -frames;
+    let perfs_n = perfs_str.and_then(|x| x.as_str().parse::<i64>().ok());
```

Comment:
Can we avoid variables like `x`? I'd prefer something descriptive rather than just a letter, outside of code for pure math.